File: app/src/main/java/com/twitli/android/twitter/tweet/Tweet.java
package com.twitli.android.twitter.tweet;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "twit_table")
public class Tweet {

    @ColumnInfo(name = "tweet_id")
    @PrimaryKey
    @NonNull
    private Long tweetId;

    @ColumnInfo(name = "name")
    private String name;

    @ColumnInfo(name = "screen_name")
    private String screenName;

    @ColumnInfo(name = "time")
    private String time;

    @ColumnInfo(name = "text")
    private String text;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getTime() {
        return time;
    }

    public void setTime(String time) {
        this.time = time;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public String getScreenName() {
        return screenName;
    }

    public void setScreenName(String screenName) {
        this.screenName = screenName;
    }

    @NonNull
    public Long getTweetId() {
        return tweetId;
    }

    public void setTweetId(@NonNull Long tweetId) {
        this.tweetId = tweetId;
    }
}

File: app/src/main/java/com/twitli/android/twitter/tweet/Tweet.kt
package com.twitli.android.twitter.tweet

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "twit_table")
class Tweet {

    @ColumnInfo(name = "tweet_id")
    @PrimaryKey
    var tweetId: Long = 0L

    @ColumnInfo(name = "name")
    var name: String? = null

    @ColumnInfo(name = "screen_name")
    var screenName: String? = null

    @ColumnInfo(name = "time")
    var time: Long = 0L

    @ColumnInfo(name = "text")
    var text: String? = null

    @ColumnInfo(name = "liked")
    var isLiked = false

    @ColumnInfo(name = "number_of_likes")
    var likes = 0

    @ColumnInfo(name = "processed")
    var processed = false
}
